### PR TITLE
Log out the number of events per key for real-time events published, …

### PR DESCRIPTION
**Pull request by iamigo**
…for perspectives and bots.

- (config/toggles.js) Define new toggle "enablePubStatsLogs" based on env var ENABLE_PUB_STATS_LOGS.
- (config.js) Define new config var pubStatsLogsIntervalMillis, default to 60000 (1 min.), override with env var PUB_STATS_LOGS_INTERVAL_MILLIS. When enablePubStatsLogs is enabled, we will write out "activity=pubStats" log lines every minute (unless admin overrides the default).
- (config/activityLog.js) Define new "activity=pubStats" log type
- (clock/scheduledJobs/pubStatsLogs.js) New scheduled job to generate the "activity=pubStats" log lines and reset the counters every <pubStatsLogsIntervalMillis>.
- (clock/index.js) When enablePubStatsLogs is enabled, add the new pubStatsLogs scheduled job to the list of jobs managed by the clock dyno.
- (realtime/constants.js) - Define a constant to use as the key to the redis cache hash object where we will store the published event counts.
- (realtime/redisSubscriber.js) - When enablePubStatsLogs is enabled, track the number of real-time events in redis cache.
- (tests/clock/pubStatsLogs.js) - Unit test the clock/pubStatsLogs.js functions

### CODE REVIEW

File: config/toggles.js [hunk 1]
```diff
@@ -60,6 +60,8 @@ const longTermToggles = {
   enableKueStatsActivityLogs:
     environmentVariableTrue(pe, 'ENABLE_KUESTATS_ACTIVITY_LOGS'),
 
+  enablePubStatsLogs: environmentVariableTrue(pe, 'ENABLE_PUB_STATS_LOGS'),
```
Comment (harshkothari410) [inline config/toggles.js]: Can you make one line comment over here about this toggle variable?

Comment (iamigo) [reply config/toggles.js]: done

File: clock/scheduledJobs/pubStatsLogs.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * Generate the logging objects, one per publish "key". Look up the pubStats
+ * hash in redis and "reset" it by deleting it; generate and return an array of
+ * pubStats activity type logging objects.
+ *
+ * @returns {Array} array of pubStats activity type logging objects
+ */
+function generateLogObjects() {
+  return Promise.all([
+    redis.hgetallAsync(PUB_STATS_HASH),
+    redis.delAsync(PUB_STATS_HASH),
```
Comment (shriramshankar) [inline clock/scheduledJobs/pubStatsLogs.js]: Your expectation is `hgetall` will run before `del`, correct ? I thought the order of execution is not guaranteed.

File: realtime/redisSubscriber.js [hunk 3]
```diff
@@ -11,54 +11,53 @@
...
-     * io emitter to send data to the browser clients.
-     */
-    emitter(io, key, parsedObj);
-  });
+  // Broadcast messages to Perspectives and Bots
+  [subBot, subPerspective].forEach((s) => {
+    s.on('message', (channel, mssgStr) => {
+      if (featureToggles.isFeatureEnabled('enableRealtimeActivityLogs')) {
+        logger.info('Size of message received by subscriber', mssgStr.length);
+      }
```
Comment (shriramshankar) [inline realtime/redisSubscriber.js]: We don't need L44. This was from the POC we did to reduce the sample size.

File: clock/scheduledJobs/pubStatsLogs.js (hunk 2 above)
Comment (iamigo) [reply clock/scheduledJobs/pubStatsLogs.js]: The order of the results is the order of the array of promises, regardless of the order of execution/completion, but it is true that there is no guarantee that the first will start before the second, so I'll pull these apart and make them sequential

**Pull request merged**